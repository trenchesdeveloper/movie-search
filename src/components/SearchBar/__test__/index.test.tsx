import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import SearchBar from "..";

const mockedSetSearchTerm = jest.fn();

// move the setSearchTerm function to the component
describe("SearchBar", () => {
  test("SearchBar component renders correctly", () => {
    render(<SearchBar setSearchTerm={mockedSetSearchTerm} />);
    const inputElement = screen.getByPlaceholderText(/Search Movie/i);

    expect(inputElement).toBeInTheDocument();
  });

  test("should be able to type into input", () => {
    render(<SearchBar setSearchTerm={mockedSetSearchTerm} />);
    const inputElement = screen.getByPlaceholderText(
      /Search Movie/i
    ) as HTMLInputElement;

    fireEvent.change(inputElement, {
      target: { value: "spider" },
    });

    expect(inputElement.value).toBe("spider");
  });

  test("SearchBar component snapshot", () => {
    const tree = renderer
      .create(<SearchBar setSearchTerm={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
