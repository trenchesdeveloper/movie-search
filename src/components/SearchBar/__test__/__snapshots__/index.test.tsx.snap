// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar SearchBar component snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100px;
  background: var(--darkGrey);
  padding: 0 20px;
}

.c1 {
  position: relative;
  max-width: var(--maxWidth);
  width: 100%;
  height: 55px;
  background: var(--medGrey);
  margin: 0 auto;
  border-radius: 40px;
  color: var(--white);
}

.c1 img {
  position: absolute;
  left: 15px;
  top: 14px;
  width: 30px;
}

.c1 input {
  font-size: var(--fontBig);
  position: absolute;
  left: 0;
  margin: 8px 0;
  padding: 0 0 0 60px;
  border: 0;
  width: 95%;
  background: transparent;
  height: 40px;
  color: var(--white);
}

.c1 input:focus {
  outline: none;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      alt="search-icon"
      src="search-icon.svg"
    />
    <input
      onChange={[Function]}
      placeholder="Search Movie"
      type="text"
      value=""
    />
  </div>
</div>
`;
